### PR TITLE
Detect if this is a merge or rebase in rebase_migration command

### PR DESCRIPTION
**Pull request by dmitrysleptsov**
Resolve https://github.com/adamchainz/django-linear-migrations/issues/80. Adds logic to check if there is a merge request in progress. If so, handle it by swapping the migration files.

### CODE REVIEW

**Comment by adamchainz** [comment]
I don't think this is the right approach. It would be better to fix django-linear-migrations to handle both cases. It's possible to detect if the current directory is a Git repository and whether there's a rebase or merge in progress with a few Git commands.

**Comment by dmitrysleptsov** [comment]
Agree, it's not the fix for the command, it's just the less risky change that allows us to modify the command. I thought it would be simpler to add changes that don't change the logic. But I can make the existing command work with both rebase and merge. I think we can check the existence of the MERGE_HEAD file in the .git directory (in the same way as the [git client does it](https://github.com/libgit2/libgit2/blob/main/src/libgit2/repository.c#L3622). 

The question for me is how far the library should go to find out the path to the .git directory?
1. We could execute something like `git rev-parse --git-dir` directly in the script to find out the path to the .git directory. But in this case we rely on the existence of `git` command.
2. We could ask the user to do it via some setting. Because I'm not sure that setting.BASE_DIR is reliable enough to use it.
And in this case, there is also a question about what we should do if there is no Git repository?

**Comment by adamchainz** [comment]
Option 2: use `git rev-parse --git-dir`. Ensure it does not crash if `git` is not available. Don't error if there is no Git repository, carry on with the existing presume-rebase behaviour.

File: src/django_linear_migrations/management/commands/rebase_migration.py [hunk 1]
```diff
@@ -184,7 +184,27 @@ def find_migration_names(max_migration_lines: list[str]) -> tuple[str, str] | No
...
+def is_merge_in_progress() -> bool:
+    try:
+        result = subprocess.run(
+            ["git", "rev-parse", "--git-dir"],
+            capture_output=True,
+            check=True,
+            text=True,
+        )
+    except (FileNotFoundError, subprocess.SubprocessError):
+        # Either `git` is not available or there is no git repository
```
Comment (dmitrysleptsov) [inline src/django_linear_migrations/management/commands/rebase_migration.py:203]: If some error should be shown to user, this function can be moved under `Command` class

**Comment by dmitrysleptsov** [comment]
It's ready for review when you have time @adamchainz . I tested it, works as expected for both `rebase` and `merge`

**Comment by adamchainz** [comment]
Great, thanks 👍. Can you update the docs to mention the automatic detection , and add a changelog note? I’ll properly review a bit later, if you can do those tasks it would help.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -2,6 +2,11 @@
 Changelog
 =========
 
+2.8.0 (2023-05-28)
+------------------
+
+* Improve `rebase_migration` command to handle both rebasing and merging of the base branch
```
Comment (adamchainz) [inline CHANGELOG.rst:8]: reStructuredText uses two backticks for code

File: CHANGELOG.rst [hunk 3]
```diff
@@ -2,6 +2,11 @@
 Changelog
 =========
 
+2.8.0 (2023-05-28)
+------------------
```
Comment (adamchainz) [inline CHANGELOG.rst:6]: headers are inserted at release time... I should add a comment to this file to clarify that.

File: tests/test_rebase_migration.py [hunk 4]
```diff
@@ -464,6 +465,104 @@ def test_works_with_three_way_merge(self):
...
+    git_command = ["git", "rev-parse", "--git-dir"]
+
+    def setUp(self) -> None:
+        subprocess_run_patch = mock.patch(
+            "django_linear_migrations.management.commands.rebase_migration"
+            ".subprocess.run"
+        )
+        self.mock_subprocess_run = subprocess_run_patch.start()
+
+        self.addCleanup(subprocess_run_patch.stop)
```
Comment (adamchainz) [inline tests/test_rebase_migration.py:507]: I'm gonna rewrite these tests to not use mocking

File: src/django_linear_migrations/management/commands/rebase_migration.py [hunk 5]
```diff
@@ -184,7 +184,28 @@ def find_migration_names(max_migration_lines: list[str]) -> tuple[str, str] | No
...
+            check=True,
+            text=True,
+        )
+    except (FileNotFoundError, subprocess.SubprocessError):
+        # Either `git` is not available or there is no git repository, fall back to
+        # default behaviour
+        return False
+
+    git_dir = result.stdout.strip()
+    return Path(git_dir).joinpath("MERGE_HEAD").exists()
```
Comment (adamchainz) [inline src/django_linear_migrations/management/commands/rebase_migration.py:208]: we can do better with `git rev-parse --verify MERGE_HEAD`, skipping getting the git dir and reading the internal files

**Review by adamchainz — CHANGES_REQUESTED** [review]
a few changes I'm going to make

**Comment by adamchainz** [comment]
I think your pull request did not allow me to push changes, so I have had to create a new one in #261.

For future ref, please ensure the box is checked: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork

**Pull request closed without merging**